### PR TITLE
Add test to prevent assignment to variable directly

### LINKED ISSUE #14290: code passes but people said it shouldnt in chat help
#### Challenge Name

Accessing Objects Properties with Variables

#### Issue Description

https://www.freecodecamp.com/challenges/accessing-objects-properties-with-variables#?solution=%0A%2F%2F%20Setup%0Avar%20testObj%20%3D%20%7B%0A%20%2012%3A%20%22Namath%22%2C%0A%20%2016%3A%20%22Montana%22%2C%0A%20%2019%3A%20%22Unitas%22%0A%7D%3B%0A%0A%2F%2F%20Only%20change%20code%20below%20this%20line%3B%0A%0Avar%20playerNumber%20%3D%2016%3B%2F%2F%20Change%20this%20Line%0AtestObj%5BplayerNumber%5D%0Avar%20player%20%3D%20%22Montana%22%3B%20%20%20%2F%2F%20Change%20this%20Line%0A

#### Browser Information

* Browser Name, Version: latest firefox browswer
* Operating System: windows 10
* Mobile, Desktop, or Tablet: desktop

#### Your Code

```js

// Setup
var testObj = {
  12: "Namath",
  16: "Montana",
  19: "Unitas"
};

// Only change code below this line;

var playerNumber = 16;// Change this Line
testObj[playerNumber]
var player = "Montana";   // Change this Line



```
#### Screenshot

image

### PR DESCRIPTION
**Pull request by Manish-Giri**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #14290 

#### Description

Added a test to ensure that the value `Montana` was not assigned to the `player` variable directly, which was incorrectly passing the challenge, as reported by OP.

Tested locally - 

screenshot 2017-04-20 13 39 44

<hr>

screenshot 2017-04-20 13 40 27

Closes #14290

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json [hunk 1]
```diff
@@ -4326,6 +4326,7 @@
         "assert(typeof player === 'string', 'message: The variable <code>player</code> should be a string');",
         "assert(player === 'Montana', 'message: The value of <code>player</code> should be \"Montana\"');",
         "assert(/testObj\\s*?\\[.*?\\]/.test(code),'message: You should use bracket notation to access <code>testObj</code>');",
+        "assert(!code.match(/player\\s*=\\s*\"\\s*Montana\\s*\"\\s*;/gi),'message: You should not assign the value of player <code>16</code> to <code>player</code> directly');",
```
Comment (dhcodes) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: I think the message should say "Montana" instead of `player`

**Comment by dhcodes** [comment]
@Manish-Giri any update on this?

**Comment by Manish-Giri** [comment]
@dhcodes Sorry, I've been away for a while now for my research stuff. Just catching up with things. I see you've made the changes in #15218 but it seems to be closed, should I be making any changes in this PR?

**Comment by dhcodes** [comment]
@Manish-Giri No worries, friend. We had just talked awhile ago about picking up "abandoned" PRs and trying to complete them as a way to accept more PRs. I didn't do the PR against your fork correctly so I closed it. You are welcome to take the line from that PR and use it here to update your code. It fixes both my suggestion and the one from @JoolsMcFly I believe.

**Comment by dhcodes** [comment]
@Manish-Giri I tested this and it works great but I just realized this line:
`You should not assign the value of player to "Montana" directly`

should be:
`You should not assign the value <code>Montana</code> to <code>player</code> directly.`

**Comment by Manish-Giri** [comment]
@dhcodes I updated it. Please review!

**Comment by dhcodes** [comment]
**LGTM**. Thanks for all the particulars. Cheers! 🌮 🦅 💯

**Pull request merged**